osu.Game.Rulesets.Taiko/Mods/TaikoModHardRock.cs

```diff
@@ -9,5 +10,13 @@ public class TaikoModHardRock : ModHardRock
     {
         public override double ScoreMultiplier => 1.06;
         public override bool Ranked => true;
+
+        private const double slider_multiplier = 1.87;
```

Comment:
As I said on discord, there *is* a 1.4 value in stable, *but* it's fudged further if hidden is *not* active.

It actually comes out that your reverse-engineered factor is ~1.33 times, or 4/3ths, larger than 1.4, which makes sense if you were testing at a 16:9 resolution (as that's the proportion of widths of a 16:9 and a 4:3 screen if they both have the same height). But honestly, between the dependence on both hidden being active and the window resolution, I'm not even sure where to *begin* porting that to lazer.

Reply:
As of right now, the 'field of view' of the bar doesn't really change with the resolution and scales to always have the same amount of notes, amount of playfield, to be visible, and it's roughly the same as it is on stable with 16:9 resoluton - so I want to say it makes sense to have it always as if it's 16:9.

Hidden has yet to be implemented, and I assume the disparity between 16:9 and 4:3 playstyles is exactly the reason - but if the field has already been scaled to be the same regardless of the resolution, Lazer's Hidden implementation will most likely be attached to that scaling and stay fixed too? I think that makes the most sense to make all players have equal conditions in the end.

Reply:
Personally I could be fine with this, *provided* that it is documented properly, for example:

```suggestion
		/// <summary>
		/// Multiplier factor applied to slider speed.
		/// </summary>
		/// <remarks>
		/// This factor is made up of two parts: the base part (1.4)
		/// and the aspect ratio adjustment (1.33). Stable applies the latter
		/// by dividing the width of the user's display by the width of a display
		/// with the same height, but a 4:3 ratio.
		/// TODO: This adjustment is only active if Hidden is off.
		/// TODO: Revisit if taiko playfield scaling ever changes away from a hard-coded 16:9
		/// (see https://github.com/ppy/osu/issues/5685).
		/// </remarks>
        private const double slider_multiplier = 1.4 * 1.33;
```

But at that point I'm not sure there's enough of the issue solved here to justify merging.

Reply:
I'm alright with adding the remarks, but I'm dropping the HD-related TODO because, as discussed on Discord, it doesn't actually look like HR is the mod that changes its behavior (from its 'lone' behavior at that aspect ratio) when both are applied, HD is. The actual note velocity with HR seems to stay the same regardless of whether HD is applied or not.

Assuming the hardcoded 16:9 stays, it wouldn't make sense for the new HD to be based around a 4:3 ratio like it is on Stable. And even if hard 16:9 doesn't stay, you can always add another, more global, not mod- but screen-width- dependent multiplication factor to the overall slider velocity that will scale off of 16:9 (the direction of making all the players see the same amount of notes regardless of the aspect ratio makes perfect sense. If only everyone being happy with that was achievable since 4:3 players are clearly used to a different experience from the 16:9 players.)